core/src/main/java/uk/gov/gchq/koryphe/impl/predicate/AreIn.java

```diff
@@ -80,7 +82,11 @@ public void setAllowedValues(final Object[] allowedValuesArray) {
 
     @Override
     public boolean test(final Collection<?> input) {
-        return null == allowedValues || allowedValues.isEmpty() || (null != input && allowedValues.containsAll(input));
+        if (null == allowedValues || allowedValues.isEmpty()) {
+            return (isNullOrEmptyAllowed);
```

Comment:
I don't think these brackets are needed

Reply:
I think this is specifically the `return (isNullOrEmptyAllowed);` should be `return isNullOrEmptyAllowed;` and not the `if` where they brackets are required.

Reply:
Ok. will change that.